From 82a6e0e13d8b868bda6666a7b8030279046ac55e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 09 Nov 2022 17:04:26 +0200
Subject: [PATCH] [SW-0] pcie_pif: reuse wfe() macro

Instead of private inline assembly implementation of sleep, reuse the
wfe() macro which is defined for any supported architecture.

Change-Id: I19910d00ab1e19a164637f35195bced25135a285
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 07bda56..da57c4b 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -268,5 +268,5 @@
 {
 	hl_flr_report_err(PIF_ERR_CPU_EXCEPTION_HALT);
 
-	__asm__ __volatile__("sleep\n\t" : : );
+	wfe();
 }
